_posts/2023-02-19-a-lean-dataops-platform.md

```diff
@@ -0,0 +1,150 @@
+
+![Flow of Data](../img/data-driven-platform-images/architecture_diagram_1.jpg)
+
+The orchestration of Fivetran and DBT is done by Dagster to ensure the dependencies between the two tools is robust. 
+
+![orchestration](../img/data-driven-platform-images/architecture_diagram_3.jpg)
+
+Our CI/CD pipeline was created with Github actions, the checks for which ensure that governance policies are maintained and Terraform is (hopefully) working as expected. For our deployments it ensures that the infrastructure is provisioned in a consistent and reliable way with the correct version control as well as change and approval processes. 
+
+These technologies enabled us to stand up a lean data platform quickly and efficiently with robust security, CI/CD infrastructure, governance checks and data testing.
+
+
+### Snowflake Governance ❄️
+The foundation of our data platform is the cloud storage technology Snowflake. Utilizing the Terraform Snowflake provider to manage, and importantly codify our Snowflake infrastructure ensures that deployments are consistent and repeatable. By using repeatable Terraform modules, we can easily create new resources, onboarding new users, have high visibility of our resources and both track and approve changes through version control in Github. The governance of the infrastructure in Snowflake was also of particular focus for us. This took its form primarily in the creation of a strong RBAC, ensuring that only authorized users have the appropriate write and/or read access to each piece of infrastructure. These access controls are granted to functional and service roles that are used by real users as well as Fivetran and DBT. The result is a secure data platform that follows the security principle of least privilege while also allowing us to get sh*t done ! 
+
+However, there are some challenges and drawbacks that we have encountered in using the Terraform Snowflake provider. This is due to the provider being a relatively new technology that is still being developed. In some cases the behavior of the provider is not as expected. For example, if you are making account level grants, using Terraform, to pieces of infrastructure and you have other business units of tenants with that same grant you can revoke the access that others have. To avoid this, it is imperative to check that all grants have the parameter `multiple_grants_enabled` set to `true`. To enforce this you can use the tool `conftest` to check your terraform configuration for this exact parameter across all our snowflake grants. The code for this test can be seen below:
```

Comment:
I would probably rearrange all the challenges into a separate section at the end

Reply:
completed 🫡